src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+    //  Data Structures
+    //////////////////////////////////////////////////////////////*/
+
+    enum State {
+        READY_FOR_BET,
+        WAITING_FOR_RANDOMNESS,
+        READY_FOR_INSURANCE,
+        READY_FOR_PLAYER_ACTION,
+        DEALER_ACTION,
+        PAYOUTS
+    }
+
+    enum Action {
+        NO_ACTION,
+        SPLIT_ACES,
+        SPLIT,
+        DOUBLE_DOWN,
+        HIT,
+        STAND
+    }
+
+    struct Game {
+        State state;
+        uint16 shoeCount;
+        uint8[] dealerCards;
+        uint16 insurance;
+        Action lastAction;
+        uint8 totalPlayerHands;
+        uint8 activePlayerHand;
+        Hand[] playerHands;
+    }
+
+    // uint8[] playerCards;
+    // uint16 betSize;
+
+    struct Hand {
+        uint8[] cards;
+        uint16 betSize;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Private Variables -- State
+    //////////////////////////////////////////////////////////////*/
+
+    Chip internal chip;
+
+    mapping(address => Game) internal games;
+
+    /*//////////////////////////////////////////////////////////////
+    //  Private Variables -- Constant
+    //////////////////////////////////////////////////////////////*/
+
+    uint16 internal constant MINIMUM_BET_SIZE = 10;
+    uint16 internal constant MAXIMUM_BET_SIZE = 100;
+
+    uint16 internal constant SHOE_STARTING_COUNT = 52 * 6;
+
+    /*//////////////////////////////////////////////////////////////
+    //  Constructor
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(Chip _chip) {
+        chip = _chip;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Views
+    //////////////////////////////////////////////////////////////*/
+
+    function getGame(address _player) public view returns (Game memory game) {
+        return games[_player];
+    }
+
+    function convertCardToValue(uint8 card) public pure returns (uint8) {
+        uint8 raw = card % 13 + 1;
+
+        if (raw >= 10) {
+            return 10;
+        } else {
+            return raw;
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Randomness
+    //////////////////////////////////////////////////////////////*/
+
+    function requestRandomness(address _player) public {}
+
+    function fulfillRandomness(address _player, bytes32 _randomness) public {
+        Game storage game = games[_player];
+
+        // Determine what to do with randomness...
+        if (game.state != State.WAITING_FOR_RANDOMNESS) {
+            // We can't fulfill randomness if not waiting for randomness.
+            revert InvalidStateTransition();
+        } else if (game.shoeCount == SHOE_STARTING_COUNT) {
+            // seed
+            // pair of aces -- player gets AA (13, 26)
+            // pair         -- player gets 77 (6, 19)
+            // other        -- player gets 54 (4, 16)
+
```

Comment:
to get three initial cards you can probably just keccak(seed + 1,2,3) and pass that into the dealCard fn